src/features/unified-share-modal/ContactRestrictionNotice.js

```diff
@@ -91,8 +105,17 @@ const ContactRestrictionNotice = ({
     let restrictionNoticeMessage;
     // Information Barrier restrictions do not allow justifications
     if (collabRestrictionType === COLLAB_RESTRICTION_TYPE_INFORMATION_BARRIER) {
-        restrictionNoticeMessage =
-            restrictionNoticeMessageMap[COLLAB_RESTRICTION_TYPE_INFORMATION_BARRIER][restrictedContactCountType];
+        if (restrictedContactCountType === SINGLE_CONTACT) {
```

Comment:
The message map approach still prevents us from writing a bunch of nested `if...else` statements. This one can't be avoided though.